packages/dds/tree/docs/main/merge-semantics.md

```diff
@@ -0,0 +1,519 @@
+While this precondition is sufficient to prevent a node of type `Foo` from being inserted in an array whose schema does not allow it,
+it is much broader than strictly necessary and will also prevent edits that would not violate the schema.
+While we plan to offer a narrower precondition in the future, the current one is tolerable because schema changes are rare.
+
+The answer to the second question will be negative if the edit does not have a precondition that would be desirable for a given scenario or has a precondition that is too permissive.
+This can often be remedied by using constraints\*.
+For example, in the scenario where Alice and Bob both want to change the color of the sticky note,
+it may be desirable to ensure that the color that is retained is always the one chosen by Alice.
+This could be achieved if Bob's edit had a precondition that the note's color must not have been concurrently changed
+(and if Alice's edit didn't have that precondition).
+`SharedTree`'s current set of edits does not include an edit with such a precondition,
+but using a constraint that ensures the original color has not been removed would likely be an acceptable alternative.
+
+\*[Constraints](#constraints) is a feature that allows transaction authors to add additional preconditions to a transaction.
+This makes understanding preconditions a prerequisite for using constraints,
+but constraints can be ignored when it comes to understanding how preconditions affect merge semantics in general.
+
+Note that the preconditions of a transaction are the union of the preconditions of its edits.
+
+### Postconditions
+
+The postconditions characterize what is guaranteed to be true about the effect of the edit provided the preconditions were satisfied.
+The term "intention" is often used in relation to an edit's postconditions: they capture what the purpose of the edit is.
+This is often expressed in terms of the resulting state after the edit is applied.
+For example, move edits guarantee that the targeted nodes end up at the specified destination.
+In other words, the intention for a move edit is to have the targeted node reside at the destination.
+
+When expressed in terms of the document state,
+these postconditions typically only hold immediately after the edit as opposed to holding indefinitely.
+In our move example, edits that come after the move
+(whether they were made concurrently or not)
+may very well cause those nodes to end up in a different location.
+
+When considering whether a given edit is suitable for a given scenario,
+the postconditions help answer the following questions:
+
+1.  Will this edit, in scenarios where it applies, have all the effects I want it to have?
+2.  Will this edit, in scenarios where it applies, have none of the effects I do not want it to have?
+
+The answer to the first question will be negative if some desirable postcondition is missing or an existing one is too narrow.
```

Comment:
Interesting that this is the only one of the four questions where you don't specify if it can be remedied. Did you want to mention "this is where we add more operations such as slice" or something? :)